dsf-gdb/org.eclipse.cdt.dsf.gdb/src/org/eclipse/cdt/dsf/gdb/service/GDBMemory.java

```diff
@@ -323,7 +331,13 @@ public int getAddressSize(IMemoryDMContext context) {
 	@Override
 	public int getAddressableSize(IMemoryDMContext context) {
 		Integer addressableSize = fAddressableSizes.get(context);
-		return (addressableSize != null) ? addressableSize.intValue() : 1;
+		assert addressableSize != null;
```

Comment:
I'm not sure regarding `assert` here, since further code handles the case under assertion as expected situation

Reply:
I'm mirroring `isBigEndian()` in the same class here. All launch sequences _should_ initialize memory data but there may be some old 3rd party launch sequences that still do not call the required method. We continue to permit this in released code (assertions disabled) but the `assert` makes it clear to extenders that they must make changes.